### PR TITLE
Stats 1 d

### PR DESCRIPTION
**Pull request by CJ-Wright**
Add function for doing more advanced statistics on 1D histograms.

### CODE REVIEW

File: skxray/core.py [hunk 1]
```diff
@@ -604,6 +605,51 @@ def bin_1D(x, y, nx=None, min_x=None, max_x=None):
...
+    # handle default values
+    if min_x is None:
+        min_x = np.min(x)
+    if max_x is None:
+        max_x = np.max(x)
+    if nx is None:
+        nx = _defaults["bins"]
+
+    # use a weighted histogram to get the bin sum
+    bins = np.linspace(start=min_x, stop=max_x, num=nx+1, endpoint=True)
```
Comment (tacaswell) [inline skxray/core.py]: We have a helper function in core to take any permutation of how to specify bin edges.  Might be worth either use that here taking in the bins array as input.

File: skxray/recip.py [hunk 2]
```diff
@@ -194,3 +197,27 @@ def hkl_to_q(hkl_arr):
     """
 
     return np.linalg.norm(hkl_arr, axis=1)
+
+
+def xpd_process_to_q(detector_size, pyfai_kwargs):
```
Comment (tacaswell) [inline skxray/recip.py]: no beam line names in function names.  maybe `calibrated_pixels_to_q`?

File: skxray/core.py [hunk 3]
```diff
@@ -604,6 +605,51 @@ def bin_1D(x, y, nx=None, min_x=None, max_x=None):
...
+    Parameters
+    ----------
+    x : array
+        position
+    y : array
+        intensity
+    stat: str or func, optional
+        statistic to be used on the binned values
+        see scipy.stats.binned_statistic
+    nx : integer, optional
```
Comment (tacaswell) [inline skxray/core.py]: state where default value comes from

File: skxray/core.py [hunk 4]
```diff
@@ -604,6 +605,51 @@ def bin_1D(x, y, nx=None, min_x=None, max_x=None):
...
+        number of bins to use
+    min_x : float, optional
```
Comment (tacaswell) [inline skxray/core.py]: What is default value?

File: skxray/core.py [hunk 5]
```diff
@@ -604,6 +605,51 @@ def bin_1D(x, y, nx=None, min_x=None, max_x=None):
...
+        Left edge of first bin
+    max_x : float, optional
+        Right edge of last bin
```
Comment (tacaswell) [inline skxray/core.py]: what is default value?

File: skxray/recip.py (hunk 2 above)
Comment (CJ-Wright) [inline skxray/recip.py]: Actually that is part of a different PR, which I think bled into this one

File: skxray/core.py [hunk 6]
```diff
@@ -604,6 +605,51 @@ def bin_1D(x, y, nx=None, min_x=None, max_x=None):
     return bins, val, count
 
 
+def statistics_1D(x, y, stat='mean', nx=None, min_x=None, max_x=None):
```
Comment (tacaswell) [inline skxray/core.py]: This needs at least a smoke test.

File: skxray/core.py (hunk 1 above)
Comment (CJ-Wright) [inline skxray/core.py]: Ok, I copied the signature from bin_1D

File: skxray/recip.py (hunk 2 above)
Comment (tacaswell) [inline skxray/recip.py]: well, the comment stands anyway!

File: skxray/core.py (hunk 6 above)
Comment (CJ-Wright) [inline skxray/core.py]: I copied/amended the one for bin_1d

**Pull request merged**